fMRIVolume/scripts/MotionCorrection.sh

```diff
@@ -181,7 +181,7 @@ function DeriveBackwards {
   # TCS becomes an array of the values from column $i in $in (derived from Var)
   TCS=($Var)
   # random is a random file name for temporary output
-  random=$RANDOM
+  random="$(mktemp --tmpdir MotionCorrectionRandom_XXXXXX.txt)"
```

Comment:
`--tmpdir` isn't a supported option on a standard MacOS installation.

Reply:
Also, if going this route, should we also make use of the `tempfiles.shlib` in this script as well?

Reply:
Hmm...I had already used `--tmpdir` elsewhere, that is unfortunate...

`tempfiles.shlib` only contains functions for cleanup, it doesn't actually generate any temporary file names currently.  If we need some compatibility workaround on Mac, then maybe it should go in there.

Reply:
On the old mac system I can easily test on, mktemp doesn't even allow generating a temporary filename with a desired file extension:

```
$ mktemp foo-XXXXXXX.txt
foo-XXXXXXX.txt
$ mktemp -t foo-XXXXXXX.txt
/tmp/foo-XXXXXXX.txt.laTQHK7E
```

This is pretty disappointing.  I guess I can write a function that uses --tmpdir on everything except mac, and fake it on mac, but this means I have to detect mac systems reliably.

Reply:
I know `mktemp` may be elegant, but why not just stick with the existing code of writing to a file named from `$RANDOM`?  We are deleting the file anyway.

Reply:
I edited it because in my test run, that folder was not writable.  Using a folder that is always writable for all temporaries is better practice than ever using the current directory, and using $RANDOM as a mechanism for trying to avoid filename collision is bad when there is already something designed for specifically that purpose.